nats.go

```diff
@@ -3149,6 +3344,16 @@ func (nc *Conn) isConnected() bool {
 	return nc.status == CONNECTED
```

Comment:
I wonder if we should consider the draining states as being connected as well, since looks like the flusher would bail once it is not in the CONNECTED state: https://github.com/nats-io/go-nats/blob/e04b7c4dc201c9e648595d08885d8fe50b5d0721/nats.go#L1904-L1908
Should the flusher be running while in the draining subs/pubs state?  Although once the subs/pubs have been drained the call to `Flush()` before closing does a `bw.Flush()` the internal buffer anyway: https://github.com/nats-io/go-nats/blob/e04b7c4dc201c9e648595d08885d8fe50b5d0721/nats.go#L2841

Reply:
Very good catch... @derekcollison Maybe we should not use a connection status, but just booleans in connection object so we don't have this kind of side effects...

Reply:
Yes good point. Since we use flush manually in the drain state it may be ok, but I will double check tomorrow and make sure. Thanks.